adminSiteServer/chartConfigHelpers.ts

```diff
@@ -0,0 +1,85 @@
+import {
+    Base64String,
+    ChartConfigsTableName,
+    DbInsertChartConfig,
+    DbRawChartConfig,
+    GrapherInterface,
+    R2GrapherConfigDirectory,
+    serializeChartConfig,
+} from "@ourworldindata/types"
+import { uuidv7 } from "uuidv7"
+import * as db from "../db/db.js"
+import {
+    saveGrapherConfigToR2,
+    saveGrapherConfigToR2ByUUID,
+} from "./chartConfigR2Helpers.js"
+
+export const retrieveChartConfigFromDbAndSaveToR2 = async (
+    knex: db.KnexReadonlyTransaction,
+    chartConfigId: Base64String,
+    r2Path?: { directory: R2GrapherConfigDirectory; filename: string }
+) => {
+    // We need to get the full config and the md5 hash from the database instead of
+    // computing our own md5 hash because MySQL normalizes JSON and our
+    // client computed md5 would be different from the ones computed by and stored in R2
+    const fullConfigMd5: Pick<DbRawChartConfig, "full" | "fullMd5"> =
```

Comment:
I think it would be better to raise an error if this is null/undefined here and not use `!` later. It shouldn't happen, but when it does, it might be easier to debug that way.